casper/src/main/scala/coop/rchain/casper/engine/GenesisCeremonyMaster.scala

```diff
@@ -52,14 +52,24 @@ object GenesisCeremonyMaster {
     /* Casper */      : Estimator: SafetyOracle: LastFinalizedBlockCalculator: LastFinalizedHeightConstraintChecker: SynchronyConstraintChecker
     /* Storage */     : BlockStore: BlockDagStorage: LastFinalizedStorage: DeployStorage: CasperBufferStorage: RSpaceStateManager
     /* Diagnostics */ : Log: EventLog: Metrics: Span] // format: on
-  (shardId: String, finalizationRate: Int, validatorId: Option[ValidatorIdentity]): F[Unit] =
+  (
+      shardId: String,
+      finalizationRate: Int,
+      validatorId: Option[ValidatorIdentity],
+      allowRespondLastFinalizedStateRequest: Boolean
```

Comment:
I strongly suggest not use `last finalized state` anywhere in the code. It's `approved state`.

Also, why there is a need for this param in Ceremony Master? It should always answer for request for state transfer